include/aspect/particle/world.h

```diff
@@ -376,14 +369,19 @@ namespace aspect
                                  const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const;
 
         /**
-         * Finds the cells containing each particle for all particles. If
-         * particles moved out of this subdomain they will be sent
-         * to their new process and inserted there. After this function call
-         * every particle is either on its current process and in its current
-         * cell, or deleted (if it could not find its new process or cell).
+         * Finds the cells containing each particle for all particles in
+         * @p particles_to_sort. If particles moved out of the local subdomain
+         * they will be sent to their new process and inserted there.
+         * After this function call every particle is either on its current
+         * process and in its current cell, or deleted (if it could not find
+         * its new process or cell).
+         *
+         * @param [in,out] particles_to_sort Multimap containing all particles
+         * that will be sorted in this function. After the function call this
+         * container is empty.
          */
         void
-        sort_particles_in_subdomains_and_cells();
+        sort_particles_in_subdomains_and_cells(std::list<std::pair<types::LevelInd, Particle<dim> > > &particles_to_sort);
```

Comment:
how long are these `list`s? `std::list` is a pretty inefficient data structure. If you know how many particles you'll need (or can estimate it via an upper bound) then using `std::vector` is typically the better choice.